### PR TITLE
[IMP] sale: handle negative qty with auto invoice

### PR DESCRIPTION
**Pull request by chevalierv**
task-2887019

### CODE REVIEW

**Review by Demesmaeker — APPROVED** [review]

File: addons/sale/models/sale_order.py [hunk 1]
```diff
@@ -1301,7 +1301,7 @@ def _has_to_be_signed(self, include_draft=False):
 
     def _has_to_be_paid(self, include_draft=False):
         transaction = self.get_portal_last_transaction()
-        return (self.state == 'sent' or (self.state == 'draft' and include_draft)) and not self.is_expired and self.require_payment and transaction.state != 'done' and self.amount_total
+        return (self.state == 'sent' or (self.state == 'draft' and include_draft)) and not self.is_expired and self.require_payment and transaction.state != 'done' and self.amount_total > 0
```
Comment (Feyensv) [inline addons/sale/models/sale_order.py:1304]: I think it's a good opportunity to indent this line :eyes:

File: addons/sale/models/payment_transaction.py [hunk 2]
```diff
@@ -175,7 +175,7 @@ def _invoice_sale_orders(self):
                 confirmed_orders._force_lines_to_invoice_policy_order()
                 invoices = confirmed_orders.with_context(
                     raise_if_nothing_to_invoice=False
-                )._create_invoices()
+                )._create_invoices(final=True)
```
Comment (Feyensv) [inline addons/sale/models/payment_transaction.py:178]: Can we have a test for that part ? :eyes:

**Review by Feyensv — COMMENTED** [review]
Please split the changes in two commits, cleanly explaining what they change/fix, cause the current commit is quite vague tbh ;) (could maybe be a FIX for the `_has_to_be_paid` change btw).
Commit messages should be self-sufficient, we shouldn't have to open the task to understand the reason of the changes.

**Review by Feyensv — COMMENTED** [review]
'stop offering payment' -> we offer the payment to the customers? that's wonderful :D

File: addons/sale/models/sale_order.py [hunk 3]
```diff
@@ -1344,11 +1344,11 @@ def _has_to_be_paid(self, include_draft=False):
...
-            (self.state == 'sent' or (self.state == 'draft' and include_draft)) and
-            not self.is_expired and
-            self.require_payment and
-            transaction.state != 'done' and
-            self.amount_total > 0
+            (self.state == 'sent' or (self.state == 'draft' and include_draft))
+            and not self.is_expired
+            and self.require_payment
+            and transaction.state != 'done'
+            and self.amount_total > 0
```
Comment (Feyensv) [inline addons/sale/models/sale_order.py:1351]: This diff should be on the first commit :D

File: addons/sale/models/sale_order.py [hunk 4]
```diff
@@ -1331,7 +1331,7 @@ def _has_to_be_signed(self, include_draft=False):
         return (self.state == 'sent' or (self.state == 'draft' and include_draft)) and not self.is_expired and self.require_signature and not self.signature
 
     def _has_to_be_paid(self, include_draft=False):
-        """Sale orders has to be paid when:
+        """A sale order has to be paid when:
```
Comment (Feyensv) [inline addons/sale/models/sale_order.py:1334]: same

**Review by Feyensv — COMMENTED** [review]
"Prior to this commit, payment is **requester** on sale orders" :eyes:

Nearly good to go :duck:

**Review by Feyensv — APPROVED** [review]
@robodoo r+ rebase-ff

**Pull request closed without merging**